vncviewer/Surface_OSX.cxx

```diff
@@ -87,10 +87,13 @@ static void render(CGContextRef gc, CGColorSpaceRef lut,
   CGContextSetBlendMode(gc, mode);
   CGContextSetAlpha(gc, alpha);
 
-  rect.origin.x = x;
-  rect.origin.y = y;
-  rect.size.width = w;
-  rect.size.height = h;
+  // adjust rect with cocoa scale factor to support HiDPI
+  const double scale_factor = cocoa_get_scale_factor();
```

Comment:
Shouldn't this just be for when rendering to the screen? Not another Surface?

And I think it also depends on FLTK version. Perhaps we need to look at the transformation matrix?